src/visudo/mod.rs

```diff
@@ -189,3 +190,26 @@ fn solve_editor_path() -> io::Result<PathBuf> {
         "cannot find text editor",
     ))
 }
+
+macro_rules! cstr {
+    ($expr:expr) => {{
+        let _: &'static [u8] = $expr;
+        debug_assert!(std::ffi::CStr::from_bytes_with_nul($expr).is_ok());
+        // SAFETY: see `debug_assert!` above
+        unsafe { CStr::from_bytes_with_nul_unchecked($expr) }
+    }};
+}
+
+fn create_temporary_dir() -> io::Result<PathBuf> {
+    let template = cstr!(b"/tmp/sudoers-XXXXXX\0").to_owned();
+
+    let ptr = unsafe { libc::mkdtemp(template.into_raw()) };
+
+    if ptr.is_null() {
```

Comment:
I think here in the error condition `ptr` should be turned back into `CString` and deallocated but since this is already the return / exit path that's not too important.